test/functional/rpc_net.py

```diff
@@ -250,7 +250,8 @@ def test_addnode_getaddednodeinfo(self):
         assert_equal(len(added_nodes), 1)
         assert_equal(added_nodes[0]['addednode'], ip_port)
         self.log.info("Check that filtering by node works")
-        self.nodes[0].addnode(node="11.22.33.44", command='add')
+        # use link-local address (169.254.0.0/16), we don't want to open connections to the internet
+        self.nodes[0].addnode(node="169.254.33.44", command='add')
```

Comment:
I checked on my network and packets to `169.254.33.44` still leave the machine and are received by the default gateway.

What about using `-connect=0` or `-proxy=127.0.0.1:1` for this particular test? One of the nodes it adds is available and it can connect to it (node2), but by reading the test I get the impression that this is not necessary. I.e. the test would still work and fulfill its purpose if it cannot connect to node2. So a setup like `-proxy=127.0.0.1:1` where it cannot connect to anybody should be ok?

Or instead of `169.254.33.44` use a random Tor address here? E.g.

```suggestion
        self.nodes[0].addnode(node="pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion", command='add')
```